README_canbus_branch.md

```diff
@@ -0,0 +1,34 @@
+# canbus Branch README
+
+The goal is to create a SimpleIoT client that will pull specific data off of a CAN bus and publish it as points, and to create a frontend that will display that data.
+
+## Documentation
+
+The `can` client works as follows:
+- One or many MCU's send CAN packages to an MPU running the `can` client
+- The `can` client is configured to accept certain PGN's and translate each SPN contained in them to a SimpleIoT Point that is sent out through NATS
+  - The `can` client will recieve multiple messages per processor cycle
+    - If they are read in by a go routine and then passed over a channel to the select loop, how do we read and pass multiple messages per processor cycle
+    - How large should the channel buffer be? We don't want to get behind and process old information in general.
```

Comment:
I generally don't use buffered channels any more as I leave all that to NATS. So most operations are reading IO or a NATs subscription, doing some processing, and then sending the data out via IO or NATS. Both IO and NATs are buffered, so our select loops should theoretically never block. Its actually quite nice. NATS will actually detect slow consumers (one that can't pull data as fast as its being pushed to it), so you still have the problem of making sure your consumers are always faster than the publishers. But, generally I've found this system nicer to work with than fixed length Go channels.